task-requests/details/script.js

```diff
@@ -200,17 +214,26 @@ async function updateTaskRequest(action, userId) {
   }
 }
 
-function getActionButton(requestor) {
+function getActionButton(requestor, taskRequest) {
+  const isSuperUser = self_user.roles['super_user'];
```

Comment:
the checks for superuser roles appear in multiple places in the codebases (e.g., in `getActionButton`, and `renderRejectButton`). This could lead to maintenance issues down the road.

We can consider abstracting role checks into a utility function like `isSuperUser()` to make it easier to handle role logic across our codebase.

Reply:
Sir. i have added a function that check the role